coral-schema/src/main/java/com/linkedin/coral/schema/avro/ToNullableSchemaVisitor.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2022 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.schema.avro;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.Lists;
+
+import org.apache.avro.Schema;
+import org.codehaus.jackson.JsonNode;
+
+
+/**
+ * Converts all the fields (including nested fields) to be nullable for a given schema
+ */
+public class ToNullableSchemaVisitor extends AvroSchemaVisitor<Schema> {
+  public static Schema visit(Schema schema) {
+    return SchemaUtilities.makeNonNullable(AvroSchemaVisitor.visit(schema, new ToNullableSchemaVisitor()));
+  }
+
+  @Override
+  public Schema record(Schema record, List<String> names, List<Schema> fields) {
+    Schema nullableSchema =
+        Schema.createRecord(record.getName(), record.getDoc(), record.getNamespace(), record.isError());
+
+    List<Schema.Field> nullableFields = Lists.newArrayListWithExpectedSize(fields.size());
+    for (int i = 0; i < fields.size(); i++) {
+      nullableFields.add(nullableField(record.getFields().get(i), fields.get(i)));
```

Comment:
if the field is already a `union` type, this will create wrap a outer union to a union, I don't think it's valid in Avro.

Reply:
Thanks for pointing it out, looks like the issue is in `makeNullable`, modified it and corresponding test avro file.